libraries/botframework-connector/src/auth/passwordServiceClientCredentialFactory.ts

```diff
@@ -0,0 +1,127 @@
+        loginEndpoint: string,
+        validateAuthority: boolean
+    ): Promise<ServiceClientCredentials> {
+        t.Record({
+            appId: t.String.withConstraint((s) => s === this.appId),
+            audience: t.String.optional(),
+            loginEndpoint: t.String.optional(),
+            validateAuthority: t.Boolean.optional(),
+        }).check({ appId, audience, loginEndpoint, validateAuthority });
+
+        const normalizedEndpoint = loginEndpoint.toLowerCase();
+        if (normalizedEndpoint.startsWith(AuthenticationConstants.ToChannelFromBotLoginUrlPrefix)) {
+            return this.appId == null
+                ? MicrosoftAppCredentials.Empty
+                : new MicrosoftAppCredentials(this.appId, this.password, undefined, audience);
+        } else if (normalizedEndpoint === GovernmentConstants.ToChannelFromBotLoginUrl) {
+            return this.appId == null
+                ? new MicrosoftAppCredentials(null, null, null, GovernmentConstants.ToChannelFromBotOAuthScope)
+                : new MicrosoftAppCredentials(
+                      this.appId,
+                      this.password,
+                      null,
+                      GovernmentConstants.ToChannelFromBotOAuthScope
+                  );
+        } else {
+            return this.appId == null
+                ? new PrivateCloudAppCredentials(null, null, null, loginEndpoint, validateAuthority)
+                : new PrivateCloudAppCredentials(
+                      this.appId,
+                      this.password,
+                      audience,
+                      normalizedEndpoint,
+                      validateAuthority
+                  );
+        }
+    }
+}
+
+class PrivateCloudAppCredentials extends MicrosoftAppCredentials {
+    private readonly _validateAuthority: boolean;
+    private __oAuthEndpoint: string;
+
+    constructor(
+        appId: string,
+        password: string,
+        oAuthScope: string,
+        oAuthEndpoint: string,
+        validateAuthority: boolean
+    ) {
+        super(appId, password, undefined, oAuthScope);
+        this.oAuthEndpoint = oAuthEndpoint;
+        this._validateAuthority = validateAuthority;
+    }
+
+    /**
+     * 
+     */
+    get validateAuthority(): boolean {
+        return this._validateAuthority;
+    }
+
+    /**
+     * Gets the OAuth endpoint to use.
+     *
+     * @returns The OAuthEndpoint to use.
+     */
+    public get oAuthEndpoint(): string {
+        return this.__oAuthEndpoint;
+    }
+
+    /**
+     * Sets the OAuth endpoint to use.
+     */
+    public set oAuthEndpoint(value: string) {
+        // aadApiVersion is set to '1.5' to avoid the "spn:" concatenation on the audience claim
+        // For more info, see https://github.com/AzureAD/azure-activedirectory-library-for-nodejs/issues/128
+        this.__oAuthEndpoint = value;
+        this.authenticationContext = new adal.AuthenticationContext(value, this.validateAuthority, undefined, '1.5');
+    }
```

Comment:
Copy of: https://github.com/microsoft/botbuilder-js/blob/ee4705c72d49a1e421d8e3c81dddfaaf8b548401/libraries/botframework-connector/src/auth/appCredentials.ts#L92-L97

With change of `true` to `this.validateAuthority`.